### PR TITLE
Fix errors: brewfile, gpg2, heroku, rvm, nvm, and add finish echo

### PR DESCRIPTION
**Pull request by fikrikarim**
So while installing I got several errors and I tried to address all the errors. I also attach logs for some error. This is my first time modifying shell script so yeaa. There's no more error when running on my machine, but this needs to be tested on a brand new machine. I also didn't create the issue first, if it's better to break this PR into several smaller ones just let me know 😃 

### 1. Brewfile error
This one is caused because of lines:
```bash
fancy_echo "Extending zsh with oh-my-zsh ..."
sh -c "$(curl -fsSL https://raw.githubusercontent.com/robbyrussell/oh-my-zsh/master/tools/install.sh)"
```
Are between the brew bundle command:
```bash
brew bundle --file=- << EOF
tap "homebrew/services"
tap "universal-ctags/universal-ctags"
...
brew "redis"
EOF
```
Running brew bundle is like running `npm install`, so it loads the configuration from Brewfile and install them. But in this case we supply the Brewfile using the `<<`. Brew bundle normally cannot run the usual shell command. So, moving the oh-my-zsh installing line to after brew bundle command can solve this issue.

[1.brewfileerror.log](https://github.com/nimbl3/laptop/files/2067548/1.brewfileerror.log)

### 2. Heroku cask error:
Before the brew bundle can run
```bash
brew "heroku/brew/heroku"
```
it needs to run
```bash
tap "heroku/brew"
```

### 3. RVM restart error #7 :
After installing RVM, we need to run
```bash
source ~/.rvm/scripts/rvm
```
so the RVM commands are available to current session. To automatically do this so the following script can working, the `source ~/.rvm/scripts/rvm` was added after installing RVM. Unfortunately, `source` is not available when running with `sh` on MacOSX, but available when running via with `bash`. So the running command is changed from
```bash
sh mac 2>&1 | tee ~/laptop.log
```
into
```bash
bash mac 2>&1 | tee ~/laptop.log
```
What I read so far is there's not much difference between `bash` and `sh` so I think this won't cause much problems.

[3.rvmerror.log](https://github.com/nimbl3/laptop/files/2067547/3.rvmerror.log)

### 4. gpg2 brew link error:
After installing brew gpg2, the following errors come up:
```bash
Error: The `brew link` step did not complete successfully
The formula built, but is not symlinked into /usr/local
Could not symlink bin/gpg
Target /usr/local/bin/gpg
already exists. You may want to remove it:
  rm '/usr/local/bin/gpg'

To force the link and overwrite all conflicting files:
  brew link --overwrite gnupg

To list all files that would be deleted:
  brew link --overwrite --dry-run gnupg
```
So the `brew link --overwrite gnupg` is added to the script file to solve the issue

[4.gpg2failed.log](https://github.com/nimbl3/laptop/files/2067546/4.gpg2failed.log)

 ### 5. Ruby error [WIP]:
After installing Ruby, there's error:
```bash
se: command not found
```
But when I try the script again somehow it doesn't pop up anymore. Needs to test this one on a brand new machine.
[5.rubyerror.log](https://github.com/nimbl3/laptop/files/2067545/5.rubyerror.log)

### 6. NVM doesn't add pathfile to `~/.zshrc`:
Even though on [its website](https://github.com/creationix/nvm#install-script) it says the script add source line to `~/.zshrc` but when I try the script it only adds to `~/.bashrc`. So a code is added to add the source for nvm to `~/.zshrc`:
```bash
if ! command -v nvm; then
  fancy_echo "Adding nvm to zshrc automatically..."
  append_to_zshrc '# nvm path
export NVM_DIR="$HOME/.nvm"
[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
[ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion'
fi
```

### 7. Error when Google Chrome is installed:
If Google Chrome is installed manually (e.g. downloading from their site) like what I did, and this code is run:
```bash
cask "google-chrome"
```
It caused error and the script won't continue to run the other command. So conditional code is installed to check whether Google Chrome is installed before running `cask "google-chrome"`.

### CODE REVIEW

File: mac [hunk 1]
```diff
@@ -151,11 +150,27 @@ cask "docker"
...
+  fancy_echo "Installing Google Chrome.."
+  brew cask install google-chrome
+  fancy_echo "Google Chrome installed."
+fi
+
+if [ ! -d "/Applications/Slack.app" ]; then
+  fancy_echo "Installing Slack.."
+  brew cask install slack
+  fancy_echo "Slack installed."
+fi
```
Comment (olivierobert) [inline mac:167]: Does it really requires to check if the app is there or not? I

**Comment by olivierobert** [comment]
@fikrikarim let's run it again on a brand new machine. There is a new comer next week so we can try that 😉

File: mac (hunk 1 above)
Comment (fikrikarim) [reply mac:167]: I installed downloaded Google Chrome first before running the script, and it just shows error and cannot continue the script. So yeah it causes problem if you have Chrome or Slack manually installed beforehand.

**Comment by fikrikarim** [comment]
@olivierobert yea sure let's test it on a brand new machine 😃

**Comment by olivierobert** [comment]
@fikrikarim Ankit run it on his machine so it seems to be good to go

File: mac [hunk 2]
```diff
@@ -151,11 +150,27 @@ cask "docker"
...
+
+brew link --overwrite gnupg
```
Comment (olivierobert) [inline mac:169]: @fikrikarim why is this required?

**Review by olivierobert — COMMENTED** [review]
One final concern

File: mac (hunk 2 above)
Comment (fikrikarim) [reply mac:169]: @olivierobert The `brew link --overwrite gnupg`?

### 4. gpg2 brew link error:
After installing brew gpg2, the following errors come up:
```bash
Error: The `brew link` step did not complete successfully
The formula built, but is not symlinked into /usr/local
Could not symlink bin/gpg
Target /usr/local/bin/gpg
already exists. You may want to remove it:
  rm '/usr/local/bin/gpg'


To force the link and overwrite all conflicting files:
  brew link --overwrite gnupg

To list all files that would be deleted:
  brew link --overwrite --dry-run gnupg
```
So the `brew link --overwrite gnupg` is added to the script file to solve the issue

[4.gpg2failed.log](https://github.com/nimbl3/laptop/files/2067546/4.gpg2failed.log)

Based on this [stackoverflow answer](https://apple.stackexchange.com/questions/281008/install-gpg-with-homebrew) I think it's safe to run them.
